sensor/common/scannerclient/grpc_client.go

```diff
@@ -212,45 +207,28 @@ func convertIndexReportToAnalysis(ir *v4.IndexReport) *ImageAnalysis {
 }
 
 func (c *v4Client) GetImageAnalysis(ctx context.Context, image *storage.Image, cfg *types.Config) (*ImageAnalysis, error) {
-	imgSHA := utils.GetSHA(image)
-	if imgSHA == "" {
-		// TODO: ROX-19576: Is the assumption that image always have SHA correct?
-		return nil, fmt.Errorf("internal: V4 scanning of an image without SHA: %s",
-			image.GetName().GetFullName())
-	}
-	hashID := fmt.Sprintf("/v4/containerimage/%s", imgSHA)
-	opts := []grpc.CallOption{grpc.WaitForReady(true)}
-	ir, err := c.GetIndexReport(ctx, &v4.GetIndexReportRequest{HashId: hashID}, opts...)
-	if err == nil {
-		return convertIndexReportToAnalysis(ir), nil
-	}
-	// Check if index was "not found", so we create.
-	if s, _ := status.FromError(err); s.Code() != codes.NotFound {
-		return nil, fmt.Errorf("get index report (hashid: %s): %w", hashID, err)
-	}
-	log.Infof("creating index report: %s", hashID)
-	scheme := "https"
+	var opts []name.Option
 	if cfg.Insecure {
-		scheme = "http"
+		opts = append(opts, name.Insecure)
 	}
-	req := &v4.CreateIndexReportRequest{
-		HashId: hashID,
-		ResourceLocator: &v4.CreateIndexReportRequest_ContainerImage{
-			ContainerImage: &v4.ContainerImageLocator{
-				Url:      fmt.Sprintf("%s://%s", scheme, image.GetName().GetFullName()),
-				Username: cfg.Username,
-				Password: cfg.Password,
-			},
-		},
+	n := fmt.Sprintf("%s/%s@%s", image.GetName().GetRegistry(), image.GetName().GetRemote(), utils.GetSHA(image))
+	ref, err := name.NewDigest(n, opts...)
+	if err != nil {
+		// TODO: ROX-19576: Is the assumption that images always have SHA correct?
```

Comment:
I don't think so anymore. That used to be the case for Sensor, but now we allow `roxctl image scan --cluster <whatever>`, so it's possible someone uses a tag instead of a digest

Reply:
My question is: Once the image metadata is pulled from the registry before it's passed to Scanner, will `utils.GetSHA()` return `""`? My guess is no, but worth checking.

Reply:
> I don't think so anymore. That used to be the case for Sensor, but now we allow `roxctl image scan --cluster <whatever>`, so it's possible someone uses a tag instead of a digest

I know this isn't @jvdm 's question, but commenting on @RTann 's comment - this is common (even before the roxctl changes).  Based on observation every non-pod workload event (deployment, cronjob, etc.) that references an image by tag will send the image with tag and no id to sensor's enricher.  Only for pod events is the SHA (ImageID) read from the containerStatus and sent to the enricher.

Reply:
@dcaravel Do you know if images are supposed to have an ID after registry metadata pulling? I think it should be guaranteed since the registry would provide the ID. I haven't checked yet, so maybe you have some pointers or knowledge about it.

Reply:
BTW, @ross is this conversation resolved?

Reply:
> @dcaravel Do you know if images are supposed to have an ID after registry metadata pulling? I think it should be guaranteed since the registry would provide the ID. I haven't checked yet, so maybe you have some pointers or knowledge about it.

A month ago I would have confidently said yes, an ID would be available from the registry metadata.  I sent you a slack message referencing an issue where this may not be the case.